internal/controller/apiextensions/composite/composition_pt.go

```diff
@@ -186,7 +186,7 @@ func (c *PTComposer) Compose(ctx context.Context, xr resource.Composite, req Com
 	if req.Environment != nil && req.Revision.Spec.Environment != nil {
 		for i, p := range req.Revision.Spec.Environment.Patches {
 			if err := ApplyEnvironmentPatch(p, xr, req.Environment); err != nil {
-				return CompositionResult{}, errors.Wrapf(err, errFmtPatchEnvironment, i)
+				return CompositionResult{}, errors.Wrapf(err, errFmtPatchEnvironment, p.Type, i, *p.FromFieldPath)
```

Comment:
`*p.FromFieldPath` would be nil (hence crash) for `CombineFromComposite` or `CombineToComposite`: https://docs.crossplane.io/v1.13/concepts/patch-and-transform/#hl-18-11

Reply:
in that case, what attribute should we use for the error string?